exercises/chapter 10/chapter10_tests.R

```diff
@@ -13,11 +13,47 @@ run_all_chunks <- function(rmd, envir = globalenv()) {
   tempR <- file.path("temp.R")  # Create temporary file in the same directory
   on.exit(unlink(tempR))
   
-  purl(input = rmd, output = tempR)
+  purl(input = rmd, output = tempR)  # Extract R code from the Rmd file
   
-  # Source the temporary R script with chdir = TRUE so that the working directory is set to the Rmd file's directory
-  source(tempR, chdir = TRUE, local = envir)
```

Comment:
@mani1381 can you please elaborate on the error this is solving?

Reply:
Dear Professor Liu,

In the previous approach, after using the purl function, the entire R chunks in the Rmd file were converted into a single R script. By running this R script as a whole, an error in one part of the code, such as a syntax error, could cause issues in other parts. This meant that a single error could potentially fail all the tests.

In the new approach, I am utilizing the structure of the file after purl, where each chunk is separated by lines that start with "## ---------". By running each chunk separately, we can ensure that the tests remain independent of each other.